corehq/apps/reports/standard/sms.py

```diff
@@ -1103,4 +1109,177 @@ class PhoneNumberReport(BaseCommConnectLogReport):
+        return table_cell
+
+    def _fmt_status(self, number):
+        if number.verified:
+            return "Verified"
+        elif number.pending_verification:
+            return "Verification Pending"
+        elif not (number.is_two_way or number.pending_verification) and PhoneNumber.get_reserved_number(number):
+            return "Already in use"
+        return "Not Verified"
+
+    def _fmt_row(self, number, owner_cache):
+        if isinstance(number, PhoneNumber):
+            return [
+                self._fmt_owner(number.owner_doc_type, number.owner_id, owner_cache),
+                number.phone_number,
+                self._fmt_status(number),
+                "Yes" if number.is_two_way else "No",
+            ]
+
+        return [
+            self._fmt_owner(number.owner_doc_type, number.owner_id, owner_cache),
+            '---',
+            '---',
+            '---',
+        ]
+
+    def _get_rows(self, paginate=True):
+        owner_cache = {}
+        if self._show_users_without_phone_numbers:
+            data = self._get_users_without_phone_numbers()
+        else:
+            data = self._get_queryset()
+
+        if paginate and self.pagination:
+            data = data[self.pagination.start:self.pagination.start + self.pagination.count]
+
+        for number in data:
+            yield self._fmt_row(number, owner_cache)
+
+    def _get_queryset(self):
+        query = PhoneNumber.objects.filter(domain=self.domain)
+
+        if self.filter_type == 'phone_number':
+            if self.phone_number_filter:
+                query = query.filter(phone_number__contains=self.phone_number_filter)
+        elif self.filter_type == 'contact':
+            if self._show_cases:
+                query = query.filter(owner_doc_type='CommCareCase', phone_number__isnull=False)
+            else:
+                query = query.filter(owner_doc_type__in=['CommCareUser', 'WebUser'])
+                if self.selected_group:
+                    query = query.filter(owner_id__in=self.user_ids_in_selected_group)
+
+                if self.has_phone_number == 'has_phone_number':
+                    query = query.filter(phone_number__isnull=False)
+
+            if self.verification_status == 'not_verified':
+                query = query.filter(pending_verification=False, verified=False)
+            elif self.verification_status == 'verification_pending':
+                query = query.filter(pending_verification=True)
+            elif self.verification_status == 'verified':
+                query = query.filter(verified=True)
+
+        return query.order_by('phone_number', 'couch_id')
+
+    def _get_users_without_phone_numbers(self):
+        query = (
+            PhoneNumber.objects.filter(domain=self.domain).filter(owner_doc_type__in=['CommCareUser', 'WebUser'])
+        )
+        users_with_phone_numbers = {x[0]: x for x in query.values_list('owner_id', 'owner_doc_type').distinct()}
```

Comment:
i think we can also simplify this a bit when we're filtering on a group by filtering on owner_id in the group's user ids